src/article2.lean

```diff
@@ -65,6 +68,22 @@ begin
     ... = k : by {simp}
 end
 
+section alternative_ex3_page43
+    lemma iso_cancel_left2 (f: A ⟶ B) [is_iso f](h k: D ⟶ A): h ≫ f = k ≫ f → h = k := 
```

Comment:
lemmas compites for the name even if inside sections....

Reply:
Sections restrict variables and parameters (and a few other things), but if you want to separate theorems/defs etc you need `namespace`